SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,906 @@
++    uint32_t dwError = 0;
++    // load tracking data
++    TDNFFilterData * pTracking = (TDNFFilterData *)userData;
++    pTracking->nPrevElement = 1;
++
++    char * pszCharData = malloc((length + 1) * sizeof(char));
++    if (!pszCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    bzero(pszCharData, (length + 1) * sizeof(char));
++    strncpy(pszCharData, content, length);
++    char * pszLintedCharData = descEscapeCharLinter(pszCharData);
++    if (!pszLintedCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    // check params 
++    if (!pTracking->nInPackage || pTracking->nPrintPackage)
++    {
++        // print to file
++        fprintf(pTracking->pbOutfile, "%s", pszLintedCharData);
++    }
++    else if (pTracking->nInPackage && !pTracking->nTimeFound)
++    {
++        // add to buffer
++        checkAndResizeBuffer(&(pTracking->pszElementBuffer), &(pTracking->nBufferMaxLen), length);
++        strcat(pTracking->pszElementBuffer, pszLintedCharData);
++    } // else do nothing (skipped package)
++
++cleanup:
++    if (pszLintedCharData)
++    {
++        free(pszLintedCharData);
++    }
++    if (pszCharData)
++    {
++        free(pszCharData);
++    }
++    return;
++error:
++    goto cleanup;
++}
++
++char * 
++TDNFFilterFile(const char * pszInFilePath, char * pszEpochTime) {
++    // vars
++    uint32_t dwError = 0;
++    TDNFFilterData * pData = NULL;
++    FILE * pbInFile = NULL;
++    FILE * pbOutFile = NULL;
++    XML_Parser bParser;
++    bzero(&bParser, sizeof(XML_Parser));
++    char pszTimeExtension[100];
++    char * pszOutFilePath = NULL;
++    time_t nEpochTime;
++    bzero(&nEpochTime, sizeof(time_t));
++    char * pszEpochEnd = NULL;
++    nEpochTime = strtoll(pszEpochTime, &pszEpochEnd, 10);
++    
++    if (nEpochTime == 0 && strcmp(pszEpochTime, "0") != 0) {
++        printf("Error converting --epochtimelimit: %s\n", pszEpochTime);
++        dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    if (pszEpochEnd != NULL && strcmp(pszEpochEnd, "") != 0) {
++        printf("The time provided to --epochtimelimit is misformatted: \'%s\' should be an epoch time\n", pszEpochTime);
++        dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    sprintf(pszTimeExtension, "-%ld.xml", nEpochTime);
++    int nOutFileLen = strlen(pszInFilePath) + strlen(pszTimeExtension);
++
++    pszOutFilePath = malloc(nOutFileLen * sizeof(char));
++    if (!pszOutFilePath)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    bzero(pszOutFilePath, nOutFileLen * sizeof(char));
++
++    // find total extension length
++    int nFileExtLen = 4; // len of ".xml"
```

Comment:
no need to add space for the extension...because I think you already included that in pszTimeExtension above in the sprintf

Reply:
OR.... you need to remove the .xml from the sprintf above

Reply:
This is still required, nFileExtLen is used to remove the arbitrary length file extension from the original input file, unrelated to the sprintf above. The file extension at a baseline will have a ".xml" but could be ".xml.gz" or ".xml.xz" or similar. This is unknown up front, so we just start with a base of 4 and check for any compression before copying the file path into our output variable.

Reply:
That said the variable name is not quite clear, I will update to use `nInFileExtLen` instead to make this more clear to the reader what it is used for.